rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/core/properties/PAResourceManagerProperties.java

```diff
@@ -238,6 +238,9 @@
      * */
     RM_SHUTDOWN_KILL_RUNTIME("pa.rm.shutdown.kill.rt", PropertyType.BOOLEAN, "true"),
 
+    /** Maximum number of iterations trying to shutdown the resource manager properly */
+    RM_SHUTDOWN_TIMEOUT("pa.rm.shutdown.timeout", PropertyType.INTEGER, "3000"),
```

Comment:
I would prefer to use a timeout in seconds rather.

In the waiting loop, you can do Thread.sleep(100), and compare the counter against RM_SHUTDOWN_TIMEOUT*10

Reply:
Done.